tests/functional/test_package.py

```diff
@@ -804,11 +804,16 @@ def _write_fake_sdist(self, setup_py, directory, ext):
             with zipfile.ZipFile(path, 'w',
                                  compression=zipfile.ZIP_DEFLATED) as z:
                 z.writestr('sdist/setup.py', setup_py)
-        else:
+        elif ext == 'tar.gz':
             with tarfile.open(path, 'w:gz') as tar:
                 tarinfo = tarfile.TarInfo('sdist/setup.py')
                 tarinfo.size = len(setup_py)
                 tar.addfile(tarinfo, io.BytesIO(setup_py.encode()))
+        elif ext == 'tar.bz2':
```

Comment:
Should there be an ``else: raise RuntimeError()`` here? or an ``assert False, "Unknown type"`` or similiar? If we ever add another ext but forget to update this, it'd be a better error message. Not a big deal either way though.

Reply:
Good call. I still need one that isn't valid since I need to generate invalid sdist formats so that I can test that the correct error is thrown when I try to read them. I added an `else` statement back to this to make an empty file with whatever extension was passed in.